src/ray/protobuf/autoscaler.proto

```diff
@@ -0,0 +1,160 @@
+syntax = "proto3";
+option cc_enable_arenas = true;
+package ray.rpc;
+
+message ResourceBundle {
+  // The fixed point resource size of the resource (granularity: 1/10000).
+  map<string, int64> resources = 1;
+}
+
+// A workload (task/actor/placement group) unfulfilled by the the cluster's internal scheduler(s).
+message QueuedResources {
+
+  // Type of placement group strategy.
+  enum PlacementStrategy {
+    // Packs Bundles into as few nodes as possible.
+    PACK = 0;
+
+    // Places Bundles across distinct nodes or processes as even as possible.
+    SPREAD = 1;
+
+    // Packs Bundles within one node. The group is not allowed to span multiple nodes.
+    STRICT_PACK = 2;
+
+    // Places Bundles across distinct nodes.
+    // The group is not allowed to deploy more than one bundle on a node.
+    STRICT_SPREAD = 3;
+  }
+
+  // The number of resources of this quantity. This should be
+  // functionally equivalent to setting quantity=1 and repeating the
+  // field many times.
+  uint64 quantity = 1;
+
+  // How badly the cluster needs the resource 0-20. 20 means the
+  // cluster cannot complete the workload without satisfying the
+  // resource (e.g. a "hard" demand). 0 means the resource is only
+  // needed for convenience.
+  int32 necessity = 2;
+
+  // Each individual resource bundle in the queued resources. For
+  // tasks or actors this is length 1.
```

Comment:
Do you mean if this QueuedResource is for a task or an actor, then this field has one entry?
In what situations will it have more than one entry?

Reply:
it would have multiple entries for a placement group. let me rephrase the comment.